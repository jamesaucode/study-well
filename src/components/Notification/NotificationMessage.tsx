import React from "react";
import styled from "styled-components";
import { Message } from "../../styles/shared";
import { NextFC } from "next";
import { IMessage } from "../../../interfaces/message";
import SVG from "react-inlinesvg";

interface WrapperProps {
  mounted?: boolean;
}
const Wrapper = styled.div<WrapperProps>`
  display: flex;
  justify-content: center;
  border-radius: 5px;
  width: 100%;
  padding: 0.75rem;
  box-sizing: border-box;
  transition: 0.5s ease-in-out opacity;
`;
const Important = styled.span`
  font-weight: 700;
`;
const SVGWrapper = styled.div`
  padding: 0 0.5rem;
`;
const renderImportant = (text: string): React.ReactElement => (
  <Important>{text}</Important>
);
const NotificationMessage: NextFC<IMessage> = ({
  success,
  message,
  id,
  // delay = 3000,
  removeNotification,
}): JSX.Element => {
  return (
    <Wrapper key={id}>
      <Message success={success}>
        <div>
          {success ? renderImportant("Success! ") : renderImportant("Oops! ")}
          {message}
        </div>
        <SVGWrapper onClick={(): void => removeNotification(id)}>
          <SVG className="small-icon" src="/static/images/close.svg" />
        </SVGWrapper>
      </Message>
    </Wrapper>
  );
};

NotificationMessage.defaultProps = {
  success: false,
  message: "No message",
};

export default React.memo(NotificationMessage);
