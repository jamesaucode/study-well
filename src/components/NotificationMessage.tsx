import React from "react";
import styled from "styled-components";
import { fadeIn, Message } from "../styles/shared";
import { NextFC } from "next";
import { MessageType } from '../../typings/message';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  border-radius: 5px;
  width: 100%;
  padding: 0.5rem;
  box-sizing: border-box;
  animation: ${fadeIn} 1s ease-out 1;
`;
const Important = styled.span`
  font-weight: 700;
`;
const renderImportant = (text: string) => <Important>{text}</Important>;
const NotificationMessage: NextFC<MessageType> = ({ success, message }) => {
  return (
    <Wrapper>
      <Message success={success}>
        {success ? renderImportant("Success! ") : renderImportant("Oops! ")}
        {message}
      </Message>
    </Wrapper>
  );
};

NotificationMessage.defaultProps = {
  success: false,
  message: "No message"
}

export default React.memo(NotificationMessage);
