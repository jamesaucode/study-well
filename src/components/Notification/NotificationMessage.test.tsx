/* eslint-disable */
import { mount } from "enzyme";
import NotificationMessage from "./NotificationMessage";

describe("Notifcation Message", () => {
    it("Render Component", () => {
        const wrapper = mount(
          <NotificationMessage message="lmao" success={true} id="random" />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});