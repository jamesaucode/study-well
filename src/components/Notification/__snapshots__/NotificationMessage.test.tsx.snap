// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifcation Message Render Component 1`] = `
<NotificationMessage
  id="random"
  message="lmao"
  success={true}
>
  <NotificationMessage__Wrapper
    key="random"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "NotificationMessage__Wrapper-sc-2ou29e-0",
            "isStatic": true,
            "lastClassName": "lluviE",
            "rules": Array [
              "display:flex;justify-content:center;border-radius:5px;width:100%;padding:0.75rem;box-sizing:border-box;transition:0.5s ease-in-out opacity;",
            ],
          },
          "displayName": "NotificationMessage__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "NotificationMessage__Wrapper-sc-2ou29e-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="NotificationMessage__Wrapper-sc-2ou29e-0 lluviE"
      >
        <style__Message
          success={true}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "style__Message-oz4l5n-4",
                  "isStatic": false,
                  "lastClassName": "jxEjaT",
                  "rules": Array [
                    "display:flex;justify-content:space-between;font-size:14px;font-weight:400;width:100%;max-width:888px;text-align:left;color:",
                    [Function],
                    ";background:",
                    [Function],
                    ";padding:14px 28px;border-radius:5px;&:hover{cursor:pointer;opacity:0.9;}",
                  ],
                },
                "displayName": "style__Message",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "style__Message-oz4l5n-4",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            success={true}
          >
            <div
              className="style__Message-oz4l5n-4 jxEjaT"
            >
              <div>
                <NotificationMessage__Important>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "NotificationMessage__Important-sc-2ou29e-1",
                          "isStatic": true,
                          "lastClassName": "gXnHLD",
                          "rules": Array [
                            "font-weight:700;",
                          ],
                        },
                        "displayName": "NotificationMessage__Important",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "NotificationMessage__Important-sc-2ou29e-1",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="NotificationMessage__Important-sc-2ou29e-1 gXnHLD"
                    >
                      Success! 
                    </span>
                  </StyledComponent>
                </NotificationMessage__Important>
                lmao
              </div>
              <NotificationMessage__SVGWrapper
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "NotificationMessage__SVGWrapper-sc-2ou29e-2",
                        "isStatic": true,
                        "lastClassName": "eOVNSC",
                        "rules": Array [
                          "padding:0 0.5rem;",
                        ],
                      },
                      "displayName": "NotificationMessage__SVGWrapper",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "NotificationMessage__SVGWrapper-sc-2ou29e-2",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <div
                    className="NotificationMessage__SVGWrapper-sc-2ou29e-2 eOVNSC"
                    onClick={[Function]}
                  >
                    <InlineSVG
                      baseURL=""
                      cacheGetRequests={false}
                      className="small-icon"
                      onLoad={[Function]}
                      src="/static/images/close.svg"
                      supportTest={[Function]}
                      uniquifyIDs={true}
                      wrapper={[Function]}
                    >
                      <span
                        className="isvg loading small-icon"
                      />
                    </InlineSVG>
                  </div>
                </StyledComponent>
              </NotificationMessage__SVGWrapper>
            </div>
          </StyledComponent>
        </style__Message>
      </div>
    </StyledComponent>
  </NotificationMessage__Wrapper>
</NotificationMessage>
`;
